### PR TITLE
Cellbender empty droplet removal

### PR DESCRIPTION
**Pull request by chaochaowong**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: subworkflows/nf-core/cellbender_empty_droplet_removal/main.nf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    ch_combined = ch_unfiltered.join(CELLBENDER_REMOVEBACKGROUND.out.barcodes)
+
+    ANNDATA_BARCODES(ch_combined)
+    ch_versions = ch_versions.mix(ANNDATA_BARCODES.out.versions)
+
+    ch_h5ad = ANNDATA_BARCODES.out.h5ad
+
+    emit:
+    h5ad = ch_h5ad
```
[the remark this answers is not in the document]
Comment (chaochaowong) [reply subworkflows/nf-core/cellbender_empty_droplet_removal/main.nf]: Got it!

File: subworkflows/nf-core/cellbender_empty_droplet_removal/main.nf [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    CELLBENDER_REMOVEBACKGROUND(ch_unfiltered)
+    ch_versions = ch_versions.mix(CELLBENDER_REMOVEBACKGROUND.out.versions)
+
+    ch_combined = ch_unfiltered.join(CELLBENDER_REMOVEBACKGROUND.out.barcodes)
+
+    ANNDATA_BARCODES(ch_combined)
+    ch_versions = ch_versions.mix(ANNDATA_BARCODES.out.versions)
+
+    ch_h5ad = ANNDATA_BARCODES.out.h5ad
```
[the remark this answers is not in the document]
Comment (chaochaowong) [reply subworkflows/nf-core/cellbender_empty_droplet_removal/main.nf]: Got it!

**Comment by chaochaowong** [comment]
Hi, @SPPearce, Thank you for your suggestions. I incorporated them into the new version. Last week there were some unrelated CI problems and then some PR run Linting hiccups. Now all tests have passed, and it is ready to be reviewed and merge. Thank you. 

To @nictru, I changed the name of this subworflow (for the scdownstream pipeline)  from `empty_droplet_removal` to `cellbender_empty_droplet_removal` to align with the naming convention. I hope it is ok with you.

**Review by SPPearce — APPROVED** [review]

**Comment by SPPearce** [comment]
Feel free to hit the merge button if you are happy.

**Pull request merged**